### DISCUSSION TITLE
Mobile UI feedback

### DISCUSSION THREAD

**Discussion by Pletoo** [sheltermanager/asm3#1514] [General]
I recently discovered the mobile UI and was planning to train our volunteers to use it until I discovered that the old UI was being replaced.

As a super user, I can appreciate the newer UI with all the features I don't use hidden away in a menu, even if it feels more like the bare bones webpage option when the internet is slow.

![New mobile UI](https://github.com/user-attachments/assets/ee902edb-d2a4-4504-bbbd-fe5b86082be4)

However I feel that the older version was more user friendly for those with limited access, and it is my preferred version.
The primary use of the mobile UI for us would be for the fosters to add media and logs to the cats in their care.
This view (with location filter set to foster instead of shelter) is just about perfect in its simplicity with easy visible access for a quick update for less tech savvy users. (Which pretty much sums up all of our fosters.)
 
![Old mobile UI](https://github.com/user-attachments/assets/ebb56693-1001-4665-a90f-60cb9716ff30)

My favorite feature is the ability to quickly give vaccines. I also really appreciate being able to add an animal. That, combined with the document scan (my 2nd favorite feature), would allow us to rapidly put an animal into the system when we are really busy (and make changes later if needed.)

The thing I miss the most between two interfaces is the quick access to the animal log without switching to the Desktop UI, especially for the fosters.

A feature I don't see us using is the signing pad, is there an option to set whether it is shown or not?

**Comment by bobintetley** [comment]
The old UI is based on a toolkit that stopped receiving security or any kind of updates in 2014. The newer UI with the burger menu follows the paradigm of modern phone apps. It seems like the assumption you are making is that your users can't learn anything!

The document scan feature is part of the new UI and was linked in the old one.

**Reply by Pletoo** [reply]
More like resistant to learning rather than unable to - I will teach them it just will take more than the 30 seconds they tend to give me. Our fosters are mostly grandmas who don't spend much time using apps on their phones (a burger menu is something they would find at a restaurant.) They are swamped at the moment (we currently have at least 75 kittens spread between 4 fosters and we are the only shelter in the area still taking in kittens.) I am going to be asking them to take pictures of these kittens and scan in their records before bringing them in to the shelter. That is a big ask.

Please don't misunderstand, I am very grateful for the mobile UI in either version. When I mentioned my favorite features that was meant to apply to both interfaces, just letting you know which ones we will use. I had initially hoped to make logs available to our fosters but we will skip that for now, I know they won't do anything that requires switching to the desktop UI.

**Comment by Pletoo** [comment]
Options set for the Desktop UI should be set also for the mobile UI. 
I know this is a resolved issue for the timeline but other options are still inconsistent.
For instance: I have removed the "good with" and "tattoo" fields for the desktop but they are still included when an animal is viewed on the mobile UI.

I would like to be able to indicate whether an additional field should be visible on the mobile UI (or maybe just which user role can see it in general) We have some fields that are not relevant to fosters. If it isn't possible to specifically choose which are shown, would it be doable to have the option of whether or not to show the additional fields tab at all on the mobile view? 

Is it intentional that the photo uploader location filter should have one for adoptions? I actually like having a list of recently adopted, but I didn't know if that was a bug or not. Also, the "woofsqueak" doesn't seem to do anything, but that might just be my phone being slow.
![Screenshot_20240802-080645](https://github.com/user-attachments/assets/0a482d82-a9be-45e9-bcf4-98c152396b63)

**Comment by bobintetley** [comment]
Yes, it was intentional that animals adopted in the last 30 days could have a photo uploaded, although I cannot remember why! I think it was something that was requested. I agree that disabled fields should not appear in the mobile UI and have added a ticket for that. I will check if there's an issue with playing sounds on mobile.

**Reply by Pletoo** [reply]
The adopted filter location does not show up on my Super User account - only on my normal user test account.

To clarify - the Woofsqueak works when the play button is used but I assume that it is supposed to play to confirm the file has been uploaded? It doesn't seem to do that but my phone can be slow in this rural area I live in.

**Reply by bobintetley** [reply]
It's only supposed to play when you hit the play button. The idea is that it makes a noise to attract the animal's attention to help with picture taking!

**Comment by Pletoo** [comment]
It is confusing to have the double dates in the vaccine section. Having an option to set whether only the date the vaccine was given is shown or whether both the due and given dates are shown would be very helpful.
For example - it would be very easy to misinterpret this and assume that vaccines have been given, when they have not.
![image](https://github.com/user-attachments/assets/45e37901-dab9-4925-b6d7-98e28efccead)

Also easy to read this incorrectly -
![image](https://github.com/user-attachments/assets/dc164368-22ac-4209-bd33-3975ce84db5a)
Could be read as:
8/1 Distemper given
8/22 Distemper booster given
8/29 Rabies given
9/12 Distemper booster given

We only need to know when the shot was given. That is all that is shown on vet records, If the vaccine has not been given there should not be a date shown.

**Comment by bobintetley** [comment]
agreed, that is confusing. Hiding the due date probably is the way to go in this situation since there's a part of the mobile UI dedicated to giving outstanding vaccinations

**Reply by Pletoo** [reply]
Thank you - it will be very helpful for our fosterers. I can grant them permission to view vaccines even if they don't have the permission to give them. Having just the given date is very intuitive for them.

**Reply by Pletoo** [reply]
Current vaccine view is even harder to read.
![image](https://github.com/user-attachments/assets/d217f172-8050-4bac-91cf-83f0450ac5c7)
The comments are unnecessary in this view (or at least should be optional whether they are shown - some might want them?)
I can see where the overdue 2nd Distemper shot might be something people want to see (again, should be optional) if it were visibly separated with a line or the red font. However the 1 year rabies booster is only there for adoption paperwork or in the rare case that cat is still at the center or has been returned - it really does not need to be in this view.

**Comment by Pletoo** [comment]
The add animal screen is not following the desktop UI options. 

These are our current options that are set:
![add options](https://github.com/user-attachments/assets/f613c898-d471-4f46-8d29-229ce14f9e85)

This is the mobile UI screen:
![add mobile UI](https://github.com/user-attachments/assets/7ad490e5-98f1-4545-98ca-4edf96ffaaf5)

I like the simplified/streamlined version for the mobile, so we don't need to add things in like the fosterer field, but we also don't need the size, unit, and location fields.

**Reply by bobintetley** [reply]
The mobile add animal doesn't obey the regular add animal options as it's supposed to be a cut down/simplified version. It would be a lot of work to duplicate the exact functionality of the regular add animal screen, which you can still use on a mobile device.

**Reply by Pletoo** [reply]
Ok - I like the streamlined version, which is why I was hoping to condense it down a little more by clearing out the fields we don't use  (size, unit, location.) I knew adding things would defeat the point of this version, I did not realize that removing things would also be difficult.

**Reply by bobintetley** [reply]
ok, fair point about removing fields, I was only thinking of adding!

**Comment by Pletoo** [comment]
Some of our Fosterers take in new animals directly without them being processed through our center first.
They don't have access to add animals or view people, they only use their phone - no computer.

I would like to create an online form that they can use at intake that I can then review and create an animal record from.
Is there any way they could access this form through the mobile UI or would we have to have our website admin add that somewhere?

ETA - The "found animal" function might work instead if it were accessible on the mobile UI, though the contact info might be tricky if they don't have access to people records.

**Reply by bobintetley** [reply]
That's a good idea, we could have a menu option to view and fill out available online forms from the mobile UI. You could put the link on your website too.

**Reply by Pletoo** [reply]
Thank you! Would there also be a way to indicate which forms should be available? (Or to mark unused ones as inactive, instead of deleting them?)

**Reply by bobintetley** [reply]
yes, it would need a mechanism to indicate the form should be available in the mobile ui
